### PR TITLE
feat: Adds a github formatter

### PR DESCRIPTION
**Pull request by isaacsanders**
I would am working to create a github actions formatter for this library, and the first step I wanted to take is to extract the other formats into modules.

### CODE REVIEW

**Comment by jeremyjh** [comment]
Hi, I'm not opposed to the refactoring but can you explain a bit more about what you are trying to accomplish? Is it returning the results in a structure understood by github actions so they appear better in PRs? And then you'll do another PR to implement an action formatter?

**Comment by isaacsanders** [comment]
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message

I have the action formatter written already, but I figured doing changes in smaller chunks was a better approach.

The link above outlines the format. It allows for inline annotations of the files from the github check run, i believe.

**Comment by isaacsanders** [comment]
Added in. Thanks for the review and promptness!

File: lib/dialyxir/formatter/github.ex [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  @behaviour Dialyxir.Formatter
+
+  @impl Dialyxir.Formatter
+  def format({_tag, {file, line}, {warning_name, _arguments}}) do
+    base_name = Path.relative_to_cwd(file)
+
+    warning = warning(warning_name)
+    string = warning.format_short(arguments)
+
+    "::error file=#{base_name},line=#{line},title=#{warning_name}::#{string}"
```
Comment (isaacsanders) [inline lib/dialyxir/formatter/github.ex:13]: @jeremyjh Would you say that everything raised is an error?

Comment (isaacsanders) [reply lib/dialyxir/formatter/github.ex]: I've moved it down to "warning"

**Comment by isaacsanders** [comment]
I've added the docs

**Review by jeremyjh — APPROVED** [review]
Looks great, thank you!

**Comment by jeremyjh** [comment]
@isaacsanders I just realized I had to release it to run CI, there are a couple of compile warnings can you please check?

**Comment by jeremyjh** [comment]
I've published to hex as 1.2.

**Comment by isaacsanders** [comment]
Thanks Jeremy! For this and for being a maintainer!

**Pull request merged**